python/ray/train/examples/pytorch/torch_fashion_mnist_example.py

```diff
@@ -48,111 +61,81 @@ def forward(self, x):
         return logits
 
 
-def train_epoch(dataloader, model, loss_fn, optimizer):
-    size = len(dataloader.dataset) // train.get_context().get_world_size()
-    model.train()
-    for batch, (X, y) in enumerate(dataloader):
-        # Compute prediction error
-        pred = model(X)
-        loss = loss_fn(pred, y)
+def train_func_per_worker(config: Dict):
+    lr = config["lr"]
+    momentum = config["momentum"]
+    epochs = config["epochs"]
+    batch_size = config["batch_size_per_worker"]
 
-        # Backpropagation
-        optimizer.zero_grad()
-        loss.backward()
-        optimizer.step()
+    # Build datasets inside worker training function
+    train_dataloader, test_dataloader = build_dataset(batch_size=batch_size)
 
-        if batch % 100 == 0:
-            loss, current = loss.item(), batch * len(X)
-            print(f"loss: {loss:>7f}  [{current:>5d}/{size:>5d}]")
+    # [1] Prepare Dataloader for distributed training
```

Comment:
We should mention that this shards the data and moves to CPU/GPU device?

Reply:
Done